packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -645,18 +662,33 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     }
   }, [connectionState, pushToast])
 
+  const disableBlurRef = useRef(false)
+
   // Reset `focusPath` when `documentId` or `params.path` changes
   useEffect(() => {
     if (ready && params.path) {
       const {path, ...restParams} = params
       const pathFromUrl = resolveKeyedPath(formStateRef.current?.value, pathFromString(path))
+
+      disableBlurRef.current = true
+
       // Reset focus path when url params path changes
       setFocusPath(pathFromUrl)
       setOpenPath(pathFromUrl)
+
+      if (focusPathRef.current !== pathFromUrl) {
+        focusPathRef.current = pathFromUrl
+        onFocusPath?.(pathFromUrl)
+      }
+
+      setTimeout(() => {
+        disableBlurRef.current = false
+      }, 0)
+
       // remove the `path`-param from url after we have consumed it as the initial focus path
       paneRouter.setParams(restParams)
```

Comment:
To avoid race conditions in Strict Mode as well as Concurrent mode we should make sure we cancel the timer on unmount:

```suggestion
      const timeout = setTimeout(() => {
        disableBlurRef.current = false
      }, 0)

      // remove the `path`-param from url after we have consumed it as the initial focus path
      paneRouter.setParams(restParams)
      
      return () => clearTimeout(timeout)
```